packages/flutter/lib/src/widgets/default_text_editing_shortcuts.dart

```diff
@@ -461,3 +483,67 @@ class DefaultTextEditingShortcuts extends StatelessWidget {
+
+    'moveLeftAndModifySelection:': ExtendSelectionByCharacterIntent(forward: false, collapseSelection: false),
+    'moveRightAndModifySelection:': ExtendSelectionByCharacterIntent(forward: true, collapseSelection: false),
+    'moveUpAndModifySelection:': ExtendSelectionVerticallyToAdjacentLineIntent(forward: false, collapseSelection: false),
+    'moveDownAndModifySelection:': ExtendSelectionVerticallyToAdjacentLineIntent(forward: true, collapseSelection: false),
+
+    'moveWordLeft:': ExtendSelectionToNextWordBoundaryIntent(forward: false, collapseSelection: true),
+    'moveWordRight:': ExtendSelectionToNextWordBoundaryIntent(forward: true, collapseSelection: true),
+    'moveToBeginningOfParagraph:': ExtendSelectionToLineBreakIntent(forward: false, collapseSelection: true),
+    'moveToEndOfParagraph:': ExtendSelectionToLineBreakIntent(forward: true, collapseSelection: true),
+
+    'moveWordLeftAndModifySelection:': ExtendSelectionToNextWordBoundaryOrCaretLocationIntent(forward: false),
+    'moveWordRightAndModifySelection:': ExtendSelectionToNextWordBoundaryOrCaretLocationIntent(forward: true),
+    'moveParagraphBackwardAndModifySelection:': ExtendSelectionToLineBreakIntent(forward: false, collapseSelection: false, collapseAtReversal: true),
+    'moveParagraphForwardAndModifySelection:': ExtendSelectionToLineBreakIntent(forward: true, collapseSelection: false, collapseAtReversal: true),
+
+    'moveToLeftEndOfLine:': ExtendSelectionToLineBreakIntent(forward: false, collapseSelection: true),
+    'moveToRightEndOfLine:': ExtendSelectionToLineBreakIntent(forward: true, collapseSelection: true),
+    'moveToBeginningOfDocument:': ExtendSelectionToDocumentBoundaryIntent(forward: false, collapseSelection: true),
+    'moveToEndOfDocument:': ExtendSelectionToDocumentBoundaryIntent(forward: true, collapseSelection: true),
+
+    'moveToLeftEndOfLineAndModifySelection:': ExpandSelectionToLineBreakIntent(forward: false),
+    'moveToRightEndOfLineAndModifySelection:': ExpandSelectionToLineBreakIntent(forward: true),
+    'moveToBeginningOfDocumentAndModifySelection:': ExpandSelectionToDocumentBoundaryIntent(forward: false),
+    'moveToEndOfDocumentAndModifySelection:': ExpandSelectionToDocumentBoundaryIntent(forward: true),
+
+    'transpose:': TransposeCharactersIntent(),
+
+    'scrollToBeginningOfDocument:': ScrollToDocumentBoundaryIntent(forward: false),
+    'scrollToEndOfDocument:': ScrollToDocumentBoundaryIntent(forward: true),
+
+    // TODO(knopp): Page Up/Down intents are missing (https://github.com/flutter/flutter/pull/105497)
+    'scrollPageUp:': ScrollToDocumentBoundaryIntent(forward: false),
+    'scrollPageDown:': ScrollToDocumentBoundaryIntent(forward: true),
+    'pageUpAndModifySelection': ExpandSelectionToDocumentBoundaryIntent(forward: false),
+    'pageDownAndModifySelection:': ExpandSelectionToDocumentBoundaryIntent(forward: true),
+
+    // Escape when there's no IME selection
```

Comment:
Period at the end.

Reply:
By "no IME selection" do you mean no field is focused or no TextEditingValue.selection is collapsed?

Reply:
I meant the IME candidate selection popup. When there is a popup TextInputContext will eat the escape key (hiding the popup). If there is no popup, TextInputContext will produce `cancelOperation:` selector, which we map to `DismissIntent`.